lib/pure/os.nim

```diff
@@ -182,14 +182,49 @@ when defined(windows):
     template wrapBinary(varname, winApiProc, arg, arg2: untyped) =
       var varname = winApiProc(newWideCString(arg), arg2)
     proc findFirstFile(a: string, b: var WIN32_FIND_DATA): Handle =
-      result = findFirstFileW(newWideCString(a), b)
+      var
+        vi: OSVERSIONINFO
+        windowsXPorLater: bool
+        windows7orLater: bool
+      vi.dwOSVersionInfoSize = cast[DWORD](vi.sizeof)
+      discard getVersionExW(addr(vi))
+      windowsXPorLater = vi.dwMajorVersion > 5 or (vi.dwMajorVersion == 5 and vi.dwMinorVersion >= 1) 
+      windows7orLater = vi.dwMajorVersion >= 6 and vi.dwMinorVersion >= 1
```

Comment:
Windows 10 has dwMajorVersion = 10 and dwMinorVersion = 0, and it can't pass through this check.